DSCResources/MSFT_xWebsite/MSFT_xWebsite.psm1

```diff
@@ -645,60 +972,44 @@ function Test-WebsiteBinding
         $Name,
 
         [Parameter(Mandatory = $true)]
-        [ValidateNotNullOrEmpty()]
         [Microsoft.Management.Infrastructure.CimInstance[]]
         $BindingInfo
     )
 
-    foreach ($Binding in $BindingInfo)
+    $InDesiredState = $true
+
+    # Ensure that desired binding information is valid (i.e. no duplicate IP address, port, and host name combinations).
+    if (-not (Test-BindingInfo -BindingInfo $BindingInfo))
     {
-        # First ensure that desired binding information is valid (i.e. no duplicate IPAddress, Port, HostName combinations).
-        if (
-            (-not (Confirm-UniqueBindingInfo -Port $Binding.Port -IPAddress $Binding.IPAddress -HostName $Binding.HostName -BindingInfo $BindingInfo)) -or
-            ($Binding.CertificateThumbprint -eq '' -and $Binding.CertificateStoreName -ne '') -or
-            ($Binding.CertificateThumbprint -ne '' -and $Binding.CertificateStoreName -eq '')
-        )
-        {
-            $ErrorId = 'WebsiteBindingInputInvalidation'
-            $ErrorCategory = [System.Management.Automation.ErrorCategory]::InvalidResult
-            $ErrorMessage = $($LocalizedData.WebsiteBindingInputInvalidationError) -f ${Name}
-            $Exception = New-Object -TypeName System.InvalidOperationException -ArgumentList $ErrorMessage
-            $ErrorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord -ArgumentList $Exception, $ErrorId, $ErrorCategory, $null
+        $ErrorId = 'WebsiteBindingInputInvalidation'
+        $ErrorCategory = [System.Management.Automation.ErrorCategory]::InvalidResult
+        $ErrorMessage = $($LocalizedData.WebsiteBindingInputInvalidationError) -f ${Name}
+        $Exception = New-Object -TypeName System.InvalidOperationException -ArgumentList $ErrorMessage
+        $ErrorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord -ArgumentList $Exception, $ErrorId, $ErrorCategory, $null
 
-            $PSCmdlet.ThrowTerminatingError($ErrorRecord)
-        }
+        $PSCmdlet.ThrowTerminatingError($ErrorRecord)
     }
 
-    # Assume bindings do not need to be updated
-    $IsUpdateRequired = $false
-
     try
     {
         $Website = Get-Website | Where-Object -FilterScript {$_.Name -eq $Name}
```

Comment:
As a though, why don't we just require website to be passed in as a parameter instead of rediscovering it here?